python/lib/dependabot/python/file_updater/pipfile_preparer.rb

```diff
@@ -130,9 +130,15 @@ def sub_auth_url(source, credentials)
 
         def config_variable_sources(credentials)
           @config_variable_sources ||=
-            credentials.
-            select { |cred| cred["type"] == "python_index" }.
-            map { |c| { "url" => AuthedUrlBuilder.authed_url(credential: c) } }
+            credentials.select { |cred| cred["type"] == "python_index" }.
+            map do |c|
+              {
+                "url" => AuthedUrlBuilder.authed_url(credential: c),
+                # Random suffix ensure names are unique. This entire source is stripped out of the final Pipfile.lock in
+                # FileUpdater#post_process_lockfile so it's okay that the names are non-deterministic.
+                "name" => "dependabot-inserted-index-#{SecureRandom.alphanumeric(5)}"
+              }
+            end
```

Comment:
Looks fine to me. It'd be nice to try a real update and make sure that these never end up leaking into a PR. I briefly read over the code, and I agree these should be stripped out but just in case.

Regarding implementation, I think I'd go with something more deterministic, like switching to `map.with_index |c, i|`, and appending the index. But doesn't matter too much I guess.

Reply:
Oh I missed your comment where you say you'll add a spec to make sure it's stripped. That's even better!